src/modules/utils.js

```diff
@@ -183,6 +183,27 @@ export default function (playerInstance, options) {
         );
     }
 
+    playerInstance.observe = () => {
+        var observer = new IntersectionObserver(
```

Comment:
We would like to keep dependencies to a minimum. Ideally we would only want dependencies for stuff like VAST, Hls.js, ... 

Do you reckon you would be able to write a function like this in pure JS?

Reply:
This API is widely available across a majority of browsers: https://caniuse.com/intersectionobserver

It's not really an external dependency.

Reply:
Ohhh haha sorry, stupid of me. In that case, just delete the lock files and I'll approve it.